lists/list_cache.go

```diff
@@ -200,7 +211,12 @@ Loop:
 		}
 	}
 
-	return factory.Create(), err
+	cache := factory.Create()
+	if cache.ElementCount() == 0 && err != nil {
+		cache = nil // don't replace existing cache
```

Comment:
This is a change from the previous behavior (I also changed a test). I think makes more sense to keep the current cache if the parsing completely failed.  
I can easily revert it if you prefer the current behavior as it's isolated to 8caf7c528408f0424b0e467d28ff3de41442ad9c.